### ISSUE TITLE
AppTheme in Library is not needed and conflicts

### ISSUE THREAD

**Issue by daniel-stoneuk** [BoardiesITSolutions/Android-MySQL-Connector#17]
**Describe the bug**
The application theme in the library conflicts and prevents build if implementation uses a theme called anything but AppTheme.

Temp fix is to add `tools:replace="android:theme"` to manifest.

**Comment by boardy** [comment]
Hi Daniel. Thanks I'll check that out, it was a while ago since I started this project so can't remember why the theme was included not sure if its something that the manifest file requires even if its not needed but I'll take a look and see if I can issue a fix.

**Comment by daniel-stoneuk** [comment]
Thanks. Am just getting started with the library, but thanks for creating the one and only connector directly for Android!

**Comment by daniel-stoneuk** [comment]
Not sure if this is an issue with the library or my use, but whatever I've tried I can't fix this error. 

UnsupportedMySQLServerException: Server Language 0xFF MySQL Server Version 8.0.21

Library version 0.40_MySQL8

Let me know if this is worth a new issue.

**Comment by boardy** [comment]
There is limited support with MySQL 8 so that might be the reason. Did you install MySQL through an installer such as windows exe or RPM for linux or is it a hosting service somewhere as I've seen some hosted DBs have some odd setups which can affect the library. 

I'll see if I can install your MySQL version and replicate and find a fix for that as well. FYI you have to use the mysql_native_password authentication mechanism as the library doesn't yet support the new sha2_caching (I think that's what its called from the top of my head ☺). 

Might be worth adding this is a new issue

**Comment by daniel-stoneuk** [comment]
I've installed it through the macOS package and via brew. I've also changed the character set to utf-8 for both tables. I'm not super familiar with the inner workings of MySQL databases so have been struggling a little! I've changed the config to mysql_native_password_authentication in the my.cnf file but still no luck. I'll keep trying and will create an issue if I have no luck. Thanks for your help!

**Comment by daniel-stoneuk** [comment]
Finally got there! Decided to switch to the mysql database that my client is using (hosting via Xampp), and changed the encoding in the my.conf like so: https://stackoverflow.com/a/3513812/5724251. Looks like this has fixed it. Sorry for hijacking this issue - let me know if you'd like me to delete all this to keep things tidy :)

Tried with utfmb4 and it worked also (https://stackoverflow.com/questions/28540772/using-utf8mb4-in-mysql) Fantastic!

**Comment by boardy** [comment]
That's great glad you got it working. I'm still going to look into the app theme and see if I can get the MySQL version you were working on originally working. I think what comes in xampp is still 5.x - not 100% sure if that's correct but would likely explain why it works OK.

**Comment by boardy** [comment]
I've created a new release, 0.41_MySQL8 which fixes the AppTheme requirement - apparently it isn't required in the manifest for libraries. 

The MySQL 8.0.21 should now work as well, there will just be a couple of setting changes you need to make in the my.cnf file such as:

```
protocol_compression_algorithms=zlib,uncompressed
```

It can also be useful to add the below line as I believe new users that are created with then default to use mysql_native_password
```
default-authentication-plugin=mysql_native_password
```

If you've already created a user you'll need to modify your user to make it use mysql_native_password with the following command:
```
ALTER USER 'user'@'%' IDENTIFIED WITH 'mysql_native_password' BY 'password'
```

Hopefully that fixes the issues you were having - let me know how it goes.

Thanks

Chris

**Comment by daniel-stoneuk** [comment]
Can confirm the AppTheme issue is fixed in 0.41_MySQL8. Thanks so much for the quick turnaround here! 

The version with XAMPP is 10.4.13-MariaDB but the library does say this is a 5.x.x version in the logs. I'll keep your above reply in mind and let you know how I get on.

Best, Dan
